### PR TITLE
Sanitycheck: misc fixes

### PR DESCRIPTION
**Pull request by nashif**
- sanitycheck: support exporting instances
- samples: clean test identifiers
- sanitycheck: support --force-platform
- sanitycheck: move export-tests handling and deduplicate code

### CODE REVIEW

**Comment by PerMac** [comment]
I didn't look in the code yet, only checked it at blackbox level. I see more test cases in the junit report after running the tests than in the exported list. To reproduce: 
`./scripts/sanitycheck --testcase-root ./tests/application_development/libcxx -p qemu_x86 --export-tests ~/scoutdir/out.txt`
out.txt has 2 lines:
Application_development,Cpp,application_development.cpp.libcxx,application_development.cpp.libcxx
Application_development,Cpp,application_development.cpp.libcxx.exceptions,application_development.cpp.libcxx.exceptions

Running the tests by: `./scripts/sanitycheck --testcase-root ./tests/application_development/libcxx -p qemu_x86 --outdir ~/scoutdir/qemu`
generates sanitycheck_report.xml with 6 test cases:
<testsuites><testsuite errors="0" failures="0" name="Sanitycheck" skip="0" tests="6" time="4.355156"><testcase classname="qemu_x86:application_development.cpp.libcxx" name="application_development.cpp.libcxx.array" time="2.213921" /><testcase classname="qemu_x86:application_development.cpp.libcxx" name="application_development.cpp.libcxx.vector" time="2.213921" /><testcase classname="qemu_x86:application_development.cpp.libcxx" name="application_development.cpp.libcxx.make_unique" time="2.213921" /><testcase classname="qemu_x86:application_development.cpp.libcxx.exceptions" name="application_development.cpp.libcxx.exceptions.array" time="2.141235" /><testcase classname="qemu_x86:application_development.cpp.libcxx.exceptions" name="application_development.cpp.libcxx.exceptions.vector" time="2.141235" /><testcase classname="qemu_x86:application_development.cpp.libcxx.exceptions" name="application_development.cpp.libcxx.exceptions.make_unique" time="2.141235" /></testsuite></testsuites>

**Comment by nashif** [comment]
@PerMac  can you please check again?

**Comment by PerMac** [comment]
@nashif The behavior is the same: 2 tests in the extracted list, 6 in sanitycheck_report.xml. Are we on the same page? I would like to know if the detailed list with all test cases that will be run on a platform can be obtained in advance. Now the extracted list and sanitycheck.xml are matched, but sanityche_report.xml has more test cases. I haven't looked in the code yet, will do this now.

**Comment by nashif** [comment]
you picked one special case where it does not work :) (and discovered a bug along the way), we were not looking for cpp file.
try something else, like tests/kernel/semaphore/semaphore

**Comment by PerMac** [comment]
You're right, but I picked this case because I've seen it was failing ;)  I saw problems with some other cases as well. I wonder if the list can be generated in a way that it matches exactly what is executed in the end on a platform. I already have seen that test cases where `#if` was used (although it is discouraged in spec) would also lead to the mismatch between the extracted list and executed tests  (e.g. /tests/arch/arm/arm_thread_swap).

**Comment by nashif** [comment]
> You're right, but I picked this case because I've seen it was failing ;) I saw problems with some other cases as well. I wonder if the list can be generated in a way that it matches exactly what is executed in the end on a platform. I

this is not going to be easy with the current implementation, some tests can only be categorized during runtime and would be marked as skipped if the filter does not match. I know this is not ideal but it needs some more thought before it is implemented.

File: scripts/sanitycheck [hunk 1]
```diff
@@ -945,6 +950,28 @@ def main():
 
         )
 
+    if (options.export_tests or options.list_tests) and options.platform:
+        if len(options.platform) > 1:
+            logger.error("When exporting tests, only platform should be specified.")
```
Comment (PerMac) [inline scripts/sanitycheck:955]: "When exporting tests, only one platform should be specified." <- probably word "one" was missing

**Review by PerMac — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]

**Pull request merged**